addons/calendar/models/calendar.py

```diff
@@ -1258,20 +1281,100 @@ def detach_recurring_event(self, values=None):
             )
 
             # do not copy the id
-            if data.get('id'):
-                del data['id']
+            data.pop('id')
             return meeting_origin.copy(default=data)
 
     @api.multi
     def action_detach_recurring_event(self):
+        """ Action used in the view to only update the virtual event we selected. It call the function to detach the virtual record and create a new from it.
+            It returns a form view in edit mode with the detached calendar event.
+        """
         meeting = self.detach_recurring_event()
         return {
             'type': 'ir.actions.act_window',
             'res_model': 'calendar.event',
             'view_mode': 'form',
             'res_id': meeting.id,
             'target': 'current',
-            'flags': {'form': {'action_buttons': True, 'options': {'mode': 'edit'}}}
+            'flags': {'form': {'action_buttons': True}, 'initial_mode': 'edit'},
+        }
+
+    @api.multi
+    def get_split_recurring_event(self):
+        """ Duplicate a calendar.event if we want to update all the virtual events from a fixed date.
+            In case of end type with count:
+                It will split the count between past events and remaining events. It will assign pas event count to original event
+                and the remaining count to the new event
+            If end type is a final_date:
+                It will set the final date of the original meeting to the selected event start date.
+                It will set the final date for the new event to the original final date.
+            Special case: If we selected the first event of the recurrence we will return the real id because there is no need to split between future and past events.
+            (update future events from the first event = update all events)
+            :return: A new record set containing the recurrence for future events.
+        """
+        real_id, meeting_origin, data = self.get_origin_meeting_and_schedule_fields()
+
+        reccurent_dates = meeting_origin._get_recurrent_date_by_event()
+        # If we want all the future event from the first event, that means we want all events and there is no need to split it.
+        real_start = pytz.UTC.localize(fields.Datetime.from_string(data['start'])).astimezone(pytz.UTC)
+        try:
+            event_position = reccurent_dates.index(real_start)
+        except:
```

Comment:
still not sure but at least add a comment to explain why :)
"this error may be raised it the user blabla"
but if it's a functional flow that is never going to work, you should prevent it by disabling the buttons or something